plotly/tools.py

```diff
@@ -2274,6 +2310,160 @@ def create_candlestick(open, high, low, close,
 
         return dict(data=data, layout=layout)
 
+    @staticmethod
+    def create_distplot(hist_data, group_labels,
+                        bin_size=1., curve_type='kde',
+                        show_hist=True, show_curve=True,
+                        show_rug=True, **kwargs):
+        """
+        BETA function that creates a distplot similar to seaborn.distplot
+
+        The distplot can be composed of all or any combination of the following
+        3 components: (1) histogram, (2) curve: (a) kernal density estimation
+        or (b)normal curve, and (3)rug plot. Additionally, multiple distplots
+        (from multiple datasets) can be created in the same plot.
+
+        :param (list) hist_data: list of histogram data, should be a list of
+            lists if multiple data sets are plotted on the same plot
+        :param (list) group_labels: list of strings of the name of each data
```

Comment:
I'll be honest, I'm *super* biased because of the IDE I use. It does full type-checking and will warn me when things are wrong. Also, I don't often have an issue writing/reading in this format though. Docstrings are just hard to get right.

This one, for example, could be more tersely *and* explicitly written:

```
:param (list[str]) group_labels: Names for each data set.
```

Here are some other factors to keep in mind:

* This function has *tons* of input parameters, which is inherently going to be confusing (not a dig on the function, btw)
* These explanations are a bit redundant.

Here's the same information in a terser (though just as informative) form:

```
        """
        BETA function that creates a distplot similar to seaborn.distplot

        The distplot can be composed of all or any combination of the following
        3 components: (1) histogram, (2) curve: (a) kernal density estimation
        or (b) normal curve, and (3) rug plot. Additionally, multiple distplots
        (from multiple datasets) can be created in the same plot.

        :param (list, list[list]) hist_data: Use list of lists to plot multiple data sets on
            the same plot.
        :param (list[str]) group_labels: Names for each data set.
        :param (float) bin_size: Size of histogram bins. Default = 1.
        :param (str) curve_type: 'kde' or 'normal'. Default = 'kde'.
        :param (bool) show_hist: Add histogram to distplot? Default = True.
        :param (bool) show_curve: Add curve to distplot? Default = True.
        :param (bool) show_rug: Add rug to distplot? Default = True.
        :param (list[str]) colors: Colors for traces.
        :param (list[str]) rug_text: Hovertext values for rug_plot.
        :return: (dict) Representation of a distplot figure.

        # other stuff
        """
```

As long as our docstrings are machine-parseable, I think we're doing it right :) If that means we switch to a different docstring format, I'd be totally fine with it. Here's a [SO question on it](http://stackoverflow.com/questions/3898572/what-is-the-standard-python-docstring-format). It seems to suggest that there are tools to *convert* from one machine-parseable docstring convention to another.